internal/client/client.go

```diff
@@ -84,18 +84,40 @@ func (c *CSAPI) JoinRoom(t *testing.T, roomIDOrAlias string, serverNames []strin
 	return GetJSONFieldStr(t, body, "room_id")
 }
 
-// SendEventSynced sends `e` into the room and waits for its event ID to come down /sync.
-// Returns the event ID of the sent event.
-func (c *CSAPI) SendEventSynced(t *testing.T, roomID string, e b.Event) string {
+func (c *CSAPI) SendEvent(t *testing.T, roomID string, e b.Event) string {
 	t.Helper()
 	c.txnID++
 	paths := []string{"_matrix", "client", "r0", "rooms", roomID, "send", e.Type, strconv.Itoa(c.txnID)}
 	if e.StateKey != nil {
 		paths = []string{"_matrix", "client", "r0", "rooms", roomID, "state", e.Type, *e.StateKey}
 	}
-	res := c.MustDo(t, "PUT", paths, e.Content)
+
+	query := make(url.Values, len(e.PrevEvents))
+	for _, prevEvent := range e.PrevEvents {
+		query.Add("prev_event", prevEvent)
+	}
+
+	if e.OriginServerTS != 0 {
+		query.Add("ts", strconv.FormatUint(e.OriginServerTS, 10))
+	}
```

Comment:
Adding in the new `ts` query param per MSC2716 spec

Reply:
This is unique for your tests only, please don't pollute the rest of the `Event` struct with special fields.

Reply:
After the MSC is merged, this is going to be part of the spec.  Would it be fine to add it here afterwards?

Reply:
I guess we could consider it not relevant to `SendEventSynced` so it doesn't need to be here afterwards.